src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -1317,17 +1317,19 @@ public void SetNonBlocking()
             // Note that there's no synchronization here, so we may set the non-blocking option multiple times
             // in a race.  This should be fine.
             //
-            if (!_nonBlockingSet)
+            if (!_isHandleNonBlocking)
             {
                 if (Interop.Sys.Fcntl.SetIsNonBlocking(_socket, 1) != 0)
                 {
                     throw new SocketException((int)SocketPal.GetSocketErrorForErrorCode(Interop.Sys.GetLastError()));
                 }
 
-                _nonBlockingSet = true;
+                _isHandleNonBlocking = true;
```

Comment:
Unrelated to the PR. I wonder if this may get reordered so the field is assigned before the socket becomes non-blocking? Or maybe the PInvoke/syscall implies this is ordered?
If it were reordered, for a concurrent caller the function may return before the socket is actually non-blocking.

Reply:
I'm pretty sure (but not 100%) that this syscall will prevent reordering here.